sci-libs/vtk/vtk-9.0.1.ebuild

```diff
@@ -0,0 +1,546 @@
+	# Note: freetype is omitted: vtk doesn't build with freetype-2.10.4
+	# Note: libharu is omitted: vtk needs an updated version (2.4.0)
+	# Note: no valid xdmf2 targets are found for system xdmf2
+	# Note: no valid target found for h5part and mpi4py
+	# ToDo: diy2 exodusII freetype h5part kissfft libharu verdict vpic
+	#	vtkm xdmf2 xdmf3 zfp
+	local -a DROPS=( doubleconversion eigen expat gl2ps glew hdf5 jpeg
+		jsoncpp libproj libxml2 lz4 lzma netcdf ogg png pugixml sqlite
+		theora tiff utf8 zlib )
+	use pegtl && DROPS+=( pegtl )
+
+	local x
+	for x in ${DROPS[@]}; do
+		ebegin "Dropping bundled ${x}"
+		rm -r ThirdParty/${x}/vtk${x} || die
+		eend $?
+	done
+	unset x
+
+	if use doc; then
+		einfo "Removing .md5 files from documents."
+		rm -f "${WORKDIR}"/html/*.md5 || die "Failed to remove superfluous hashes"
+		sed -e "s|\${VTK_BINARY_DIR}/Utilities/Doxygen/doc|${WORKDIR}|" \
+			-i Utilities/Doxygen/CMakeLists.txt || die
+	fi
+
+	cmake_src_prepare
+
+	if use cuda; then
+		cuda_add_sandbox -w
+		cuda_src_prepare
+	fi
+
+	if use test; then
+		ebegin "Copying data files to ${BUILD_DIR}"
+		mkdir -p "${BUILD_DIR}/ExternalData" || die
+		pushd "${BUILD_DIR}/ExternalData" >/dev/null || die
+		ln -sf "../../VTK-${PV}/.ExternalData/README.rst" . || die
+		ln -sf "../../VTK-${PV}/.ExternalData/SHA512" . || die
+		ln -sf "../../VTK-${PV}/.ExternalData/MD5" . || die
+		popd >/dev/null || die
+		eend "$?"
+	fi
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DBUILD_SHARED_LIBS=ON
+
+# defaults for some variables to consider
+#		-DVTK_ANDROID_BUILD=OFF
+#		-DVTK_BUILD_COMPILE_TOOLS_ONLY=OFF
+#		-DVTK_ENABLE_LOGGING=ON
+#		-DVTK_ENABLE_REMOTE_MODULES=ON
+#		-DVTK_INSTALL_SDK=ON
+#		-DVTK_IOS_BUILD=OFF
+#		-DVTK_LEGACY_REMOVE=OFF
+#		-DVTK_LEGACY_SILENT=OFF
+#		-DVTK_WHEEL_BUILD=OFF
+
+		-DVTK_BUILD_ALL_MODULES=$(usex all-modules ON OFF)
+		# we use the pre-built documentation and install these with USE=doc
+		-DVTK_BUILD_DOCUMENTATION=OFF
+		# FIXME: need to unmask examples first
+#		-DVTK_BUILD_EXAMPLES=$(usex examples ON OFF)
+		-DVTK_BUILD_EXAMPLES=ON
+
+		-DVTK_ENABLE_KITS=$(usex kits ON OFF)
+		# default to ON: USE flag for this?
+		-DVTK_ENABLE_REMOTE_MODULES=OFF
+
+		-DVTK_DATA_STORE="${S}/.ExternalData"
+
+		# Use upstream default, where USE flags are not given.
+		# Passing "DONT_WANT" will restrict building of modules from
+		# those groups and will severly limit the built libraries.
+		# Exceptions are MPI, where the default is "DONT_WANT" and
+		# StandAlone using "WANT".
+		-DVTK_GROUP_ENABLE_Imaging=$(usex imaging "WANT" "DEFAULT")
+		-DVTK_GROUP_ENABLE_Qt=$(usex qt5 "WANT" "DEFAULT")
+		-DVTK_GROUP_ENABLE_Rendering=$(usex rendering "WANT" "DEFAULT")
+		-DVTK_GROUP_ENABLE_StandAlone="WANT"
+		-DVTK_GROUP_ENABLE_Views=$(usex views "WANT" "DEFAULT")
+		-DVTK_GROUP_ENABLE_Web=$(usex web "WANT" "DEFAULT")
+
+		-DVTK_MODULE_ENABLE_VTK_vtkm="WANT"
+		-DVTK_MODULE_ENABLE_VTK_AcceleratorsVTKm="WANT"
+
+		# fails to build with external freetype
```

Comment:
bug? ;)

Reply:
Can't remeber exactly and don't have any of the log files saved. I need to investigate the failure reason first.

Reply:
The patch for 8.2.0 worked, although with some fuzz. Don't know why I didn't recognize this patch earlier.